kbase-extension/static/kbase/js/kbaseNarrative.js

```diff
@@ -1250,17 +1252,39 @@ define([
 
     /**
      * This inserts a new bulk import cell below the currently selected cell.
-     * Its input is a map from object type to a list of files to be uploaded of that type.
-     * @param {object} typesToFiles keys = types, values = list of files mapped to that type
+     * Its input is a map from object type to a the files to be uploaded and the app
+     * used to process them.
+     * {
+     *   fileType: {
+     *     appId: string,
+     *     files: array of files
+     *   }
+     * }
+     * This returns a Promise that resolves into the cell that was created.
+     * @param {object} bulkInput keys = type ids, values = an object with properties
+     *  - appId - the app id to use for that file type (to be used in fetching the spec)
+     *  - files - array of files to import with that file type
      */
-    Narrative.prototype.insertBulkImportCell = function (typesToFiles) {
+    Narrative.prototype.insertBulkImportCell = function (bulkInput) {
```

Comment:
The big change here is that this now returns a Promise that resolves into the new cell. That Promise includes fetching the app specs, and additional info that the info tab currently needs.